sunpy/coordinates/transformations.py

```diff
@@ -76,108 +63,155 @@ def _observers_are_equal(obs_1, obs_2, string_ok=False):
 # =============================================================================
 
 
-@frame_transform_graph.transform(FunctionTransform, HeliographicStonyhurst,
-                                 HeliographicCarrington)
+def _transform_obstime(frame, obstime):
+    """
+    Transform a frame to a new obstime using the appropriate loopback transformation
+    """
+    if frame.obstime is None:
+        raise ValueError("To perform this transformation the coordinate"
+                         " Frame needs an obstime Attribute")
+
+    # If obstime is None or the obstime matches, nothing needs to be done
+    if obstime is None or np.all(frame.obstime == obstime):
+        return frame
+
+    if isinstance(frame, BaseCoordinateFrame):
+        frame_class = frame.__class__
+    else:  # Assume that it is like SkyCoord and contains the frame
+        frame_class = frame.frame.__class__
+
+    # Transform to the new obstime using the appropriate loopback transformation
+    new_frame = frame_class(obstime=obstime)
+    return frame.transform_to(new_frame)
+
+
+def _rotation_matrix_hgs_to_hgc(obstime):
+    """
+    Return the rotation matrix from HGS to HGC at the same observation time
+    """
+    if obstime is None:
+        raise ValueError("To perform this transformation the coordinate"
+                         " Frame needs an obstime Attribute")
+
+    # Import here to avoid a circular import
+    from .sun import L0
+
+    # Rotation is only in longitude, so only around the Z axis
+    return rotation_matrix(-L0(obstime), 'z')
+
+
+@frame_transform_graph.transform(FunctionTransformWithFiniteDifference,
+                                 HeliographicStonyhurst, HeliographicCarrington)
 def hgs_to_hgc(hgscoord, hgcframe):
     """
-    Transform from Heliographic Stonyhurst to Heliograpic Carrington.
+    Convert from Heliographic Stonyhurst to Heliographic Carrington.
     """
-    if hgcframe.obstime is None or np.any(hgcframe.obstime != hgscoord.obstime):
-        raise ValueError("Can not transform from Heliographic Stonyhurst to "
-                         "Heliographic Carrington, unless both frames have matching obstime.")
+    # First transform the HGS coord to the HGC obstime
+    int_coord = _transform_obstime(hgscoord, hgcframe.obstime)
```

Comment:
I am not seeing much meaning in `int_coord` as a variable name?

Reply:
It's short for "intermediate coordinate".  I use that and `int_frame` quite a bit in my new-frames PR (although that use may go down now that I came up with the `_transform_obstime()` approach).